### ISSUE TITLE
issues after usage

### ISSUE THREAD

**Issue by sadraskol** [thomaslule/fixed-footnotes#4]
So i've been using the lib for my [blog](https://sadraskol.com/posts/why-null-leads-to-bad-implementations), but some problem cropped up:

1) links are removed/added to the DOM without any generic event associated. Meaning it's getting very hard to manage events on the content in the footer. A solution i can see: create an event for every changes on the footnotes.
2) The current html wrapper is difficult to stylize. Most specifically to create gutters, etc. look at the css I used to do so:
```sass
.fixed-footnotes-container {
  @media (min-width: $tablet) {
    display: grid;
    grid-template-columns: 25% auto;

    &:before {
      content: '';
      grid-column: 1;
      grid-row: 1 / 3;
    }
  }
}
```
Using grid is pretty hacky in this situation. I would suggest to use a template that would look like:
```html
<div class="fixed-footnotes-container">
  <ul>
     <li class="fixed-footnotes-note"></li>
  </ul>
</div>
```
Which would allow to use `:before` and `:after` with a better flexibility.

Cheers!

**Comment by thomaslule** [comment]
About 1: right now the view is refreshed on scroll and resize but there are no way to know if the visible references *actually* changed from a user perpective. For example, if a reference is displayed and there is a little scroll that keeps the reference on screen, what is actually happening is:

* we empty the references container
* we add again the same reference

Will it solve your problem if we send an event on all those refresh? (this function is throttled so the event won't be sent more than once every 200ms)

---

About 2: Right, a list makes more sense. I'll do that.

**Comment by thomaslule** [comment]
Fixed in v1.1.0
